### PR TITLE
Add unit tests for info.bitrich.xchangestream.util.BookSanityChecker

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
Hi,

I've analysed your code base and noticed that `info.bitrich.xchangestream.util.BookSanityChecker` is not fully tested.

I've written some tests for the functions in this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/products).

Hopefully these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+import static java.lang.String.format;
+
+public class BookSanityCheckerTest {
+
+  @Rule
+  public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testNoOrders() throws Exception {
```
Comment (paulkennethkent) [inline xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java]: remove unnecessary `throws Exception`

There are a few more below.

File: xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+
+import static java.lang.String.format;
+
+public class BookSanityCheckerTest {
+
+  @Rule
+  public final ExpectedException thrown = ExpectedException.none();
```
Comment (paulkennethkent) [inline xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java]: I might be missing something here but I can't see that you are throwing an exceptions

File: xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java (hunk 1 above)
Comment (JohnLBergqvist) [reply xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java]: Fixed.

File: xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java (hunk 2 above)
Comment (JohnLBergqvist) [reply xchange-stream-core/src/test/java/info/bitrich/xchangestream/util/BookSanityCheckerTest.java]: Fixed.

**Review by paulkennethkent — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request closed without merging**